test/hotspot/jtreg/serviceability/jvmti/events/Breakpoint/breakpoint01/libbreakpoint01.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2003, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "jni.h"
+#include "jni_md.h"
+#include "jvmti.h"
+
+#include "jvmti_common.h"
+
+extern "C" {
+
+#define STATUS_FAILED 2
+#define PASSED 0
+
+
+#define METH_NUM 4
+static const char *METHODS[][2] = {
+    {"bpMethod", "()V"},
+    {"bpMethod2", "()I"},
+    {"bpMethodV", "()V"},
+    {"bpMethod2V", "()I"}
+};
+
+static const jboolean METHODS_ATTRS[METH_NUM] = {JNI_FALSE, JNI_FALSE, JNI_TRUE, JNI_TRUE};
+
+
+static const char *CLASS_SIG = "Lbreakpoint01;";
+
+static const char *THREAD_NAME = "breakpoint01Thr";
+
+static volatile int bpEvents[METH_NUM];
+static volatile jint result = PASSED;
+static jvmtiEnv *jvmti = NULL;
+static jvmtiEventCallbacks callbacks;
+
+static volatile int callbacksEnabled = NSK_TRUE;
+static jrawMonitorID agent_lock;
+
+static void initCounters() {
+  int i;
+
+  for (i = 0; i < METH_NUM; i++)
+    bpEvents[i] = 0;
+}
```

Comment:
As I understand all new tests are C++ based.
So, I'd suggest to always use C++ style of loops:
  for (int i = 0; i < METH_NUM; i++)

Reply:
fixed in all ported tests were applicable